std/utf.d

```diff
@@ -4385,4 +4386,12 @@ if (isSomeChar!C)
     "𐐷".byUTF!char().equal([0xF0, 0x90, 0x90, 0xB7]);
     "𐐷".byUTF!wchar().equal([0xD801, 0xDC37]);
     "𐐷".byUTF!dchar().equal([0x00010437]);
+
+    try
+    {
+        "hello0xFFbetty".byUTF!(dchar, UseReplacementDchar.yes).equal("hello");
```

Comment:
this should be an `assert[Not]Thrown`. That whole unittest looks like it does nothing, I assume that the above (not including this one) are supposed to no throw? 
This one explicit uses `ReplacementDchar` so it would seem to me that it should not be throwing, why the catch?

Reply:
> That whole unittest looks like it does nothing

If it doesn't throw, the `equal()` will assert.

> why the catch?

So the unittest run won't exit.

Reply:
> the equal() will assert.

Really??? That is `std.algorithm.comparison.equal` we are talking about here, right? Why on earth would that assert? 

For reference, every other unittest in this file e.g. https://github.com/dlang/phobos/blob/61200ae059ea3d5a7fd068aef9d1a425e0f7e7f9/std/utf.d#L2934 is of the form `rangeOrString.toUTFBlah.equal(["expected output"])` except for this one. Why this is different I have no idea.

Regardless, If this is expected to throw then it should use, `assertThrown`, if not it should use `assertNotThrown`.

Reply:
> equal() will assert.

Not without an `assert`.

> the unittest run won't exit.

Isn't this the point of a unittest?

Reply:
Ah, you're right.

Reply:
> rangeOrString.toUTFBlah.equal(["expected output"]) except for this one. Why this is different I have no idea.

They need to use `assert` too.

Reply:
No, they already do that, I forgot to include that, whoops! Its only the one just above this that doesn't.